### PR TITLE
create event details function and query and test it

### PR DESCRIPTION
**Pull request by MostafaBalousha123**
relates #12

### CODE REVIEW

File: src/controllers/EventsController.ts [hunk 1]
```diff
@@ -1,85 +1,91 @@
...
+    res.status(200).json({ message: Message.SUCCESS, data: allEvents })
   }
 
-  // for getting just on element of data (like getting just one event may be in event details)
+  // for getting just on element of data
+  // (like getting just one event may be in event details)
   public static async show (req: Request, res: Response) {
-    // code here
+    const { id } = req.params
+    const eventDetails:any = await Event.findOne({
```
Comment (Ibrahim-Jarada) [inline src/controllers/EventsController.ts:55]:
```suggestion
    const eventDetails = await Event.findOne({
```

File: src/controllers/EventsController.ts [hunk 2]
```diff
@@ -1,85 +1,91 @@
...
 import querySchema from '../validation/addEventValidate'
 import { Message } from '../config/messages'
 import { Event, User } from '../db'
 import { Op } from 'sequelize'
+import CustomError from '../helpers/CustomError'
 
 export default class EventsController {
   // for getting all data
-  public static async index (req: Request, res: Response, next: NextFunction) {
+  public static async index (req: Request, res: Response) {
```
Comment (Ibrahim-Jarada) [inline src/controllers/EventsController.ts:10]: You may need to specify the return type of the function
```suggestion
  public static async index (req: Request, res: Response):Promise<void> {
```

File: src/controllers/EventsController.ts [hunk 3]
```diff
@@ -1,85 +1,91 @@
...
+    await querySchema.validateAsync(req.body)
+    const event = await Event.create({
+      name: data.name,
+      description: data.description,
+      img: data.img,
+      status: data.status,
+      startTime: data.startTime,
+      endTime: data.endTime,
+      longitude: data.longitude,
+      latitude: data.latitude
```
Comment (Ibrahim-Jarada) [inline src/controllers/EventsController.ts:83]: Just my preference
```suggestion
    await querySchema.validateAsync(req.body)
    const {
      name,
      description,
      img,
      status,
      startTime,
      endTime,
      longitude,
      latitude
    } = req.body
    const event = await Event.create({
      name,
      description,
      img,
      status,
      startTime,
      endTime,
      longitude,
      latitude
    })
```

**Review by Ibrahim-Jarada — COMMENTED** [review]
I notice some features of Eslint not working well and I believe its from  configuration.
consider the configuration below.
After that some Eslint errors will popup you will need to fix the reasonable ones and maybe you will need to turn off some weird rules.
```
 "extends": [
        "standard-with-typescript",
        "plugin:@typescript-eslint/recommended",
        "plugin:@typescript-eslint/eslint-recommended"
    ],
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "ecmaVersion": 11,
        "sourceType": "module",
        "project": "./tsconfig.json"
    },
```

File: .eslintrc.json [hunk 4]
```diff
@@ -16,6 +16,6 @@
         "@typescript-eslint"
     ],
     "rules": {
-        "max-len": ["error", { "code": 100 }]
+        "max-len": ["error", { "code": 200 }]
```
Comment (aaamra) [inline .eslintrc.json:19]: i think 100 chars is enough
```suggestion
        "max-len": ["error", { "code": 100 }]
```

File: client/.eslintrc.json [hunk 5]
```diff
@@ -27,7 +27,7 @@
         "semi": [1, "never"],
         "no-nested-ternary": 0,
         "no-use-before-define": "off",
-        "max-len": ["error", { "code": 100 }],
+        "max-len": ["error", { "code": 200 }],
```
Comment (aaamra) [inline client/.eslintrc.json:30]:
```suggestion
        "max-len": ["error", { "code": 100 }],
```

**Review by aaamra — CHANGES_REQUESTED** [review]

**Review by aaamra — APPROVED** [review]

**Review by shamskhodary — APPROVED** [review]

File: src/controllers/EventsController.ts [hunk 6]
```diff
@@ -1,85 +1,116 @@
...
+        ['startTime', 'ASC']
+      ]
+    })
+    res.json({ message: Message.SUCCESS, data: allEvents })
   }
 
-  // for getting just on element of data (like getting just one event may be in event details)
+  // for getting just on element of data
+  // (like getting just one event may be in event details)
   public static async show (req: Request, res: Response) {
```
Comment (Ibrahim-Jarada) [inline src/controllers/EventsController.ts:67]:
```suggestion
  public static async show (req: Request, res: Response):Promise<void> {
```

File: src/controllers/EventsController.ts [hunk 7]
```diff
@@ -1,85 +1,116 @@
...
+    if (!eventDetails) throw new CustomError(Message.NOTFOUND, 404)
+    res.status(200).json({
+      message: Message.SUCCESS,
+      data: eventDetails.dataValues
+    })
   }
 
   // for storing new data
-  public static async store (req: Request, res: Response, next: NextFunction) {
+  public static async store (req: Request, res: Response) {
```
Comment (Ibrahim-Jarada) [inline src/controllers/EventsController.ts:86]:
```suggestion
  public static async store (req: Request, res: Response):Promise<void> {
```

**Review by Ibrahim-Jarada — APPROVED** [review]

**Review by Bakeza — APPROVED** [review]

**Review by SaifHayek — APPROVED** [review]

**Pull request merged**